src/idl/src/tree.c

```diff
@@ -772,6 +772,59 @@ uint32_t idl_bound(const void *node)
   return 0u;
 }
 
+bool idl_is_optional(const void *node)
+{
+  if (idl_is_declarator(node)) {
+    const idl_node_t *parent = idl_parent(node);
+    if (idl_is_member(parent))
+      return idl_is_optional(parent);
+  } else {
+    idl_annotation_appl_t *annotation_appl = NULL;
+    IDL_FOREACH(annotation_appl, ((const idl_node_t*)node)->annotations) {
+      if (0 != strcmp(idl_identifier(annotation_appl->annotation), "optional"))
+        continue;
+
+      if (annotation_appl->parameters) {
+        if (annotation_appl->parameters->const_expr
+         && idl_is_literal(annotation_appl->parameters->const_expr)
+         && idl_mask(annotation_appl->parameters->const_expr) == (IDL_LITERAL | IDL_BOOL))
+          return ((const idl_literal_t*)annotation_appl->parameters->const_expr)->value.bln;
+      } else {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+const idl_literal_t *idl_has_default(const void *node)
+{
+  const struct idl_annotation_appl *annotation_appl = NULL;
+
+  IDL_FOREACH(annotation_appl, ((const idl_node_t*)node)->annotations) {
+    if (0 != strcmp(idl_identifier(annotation_appl->annotation), "default"))
+      continue;
+
+    if (annotation_appl->parameters
+     && annotation_appl->parameters->const_expr
+     && idl_is_literal(annotation_appl->parameters->const_expr))
+      return (const idl_literal_t*)annotation_appl->parameters->const_expr;
+  }
+
+  return NULL;
+}
+
+bool idl_is_default(const void *node)
+{
+  const idl_literal_t *lit = idl_has_default(node);
+
+  if (lit
+   && idl_mask(lit) == (IDL_LITERAL | IDL_ANY))
+    return lit->value.bln;
```

Comment:
Does this also work if an explicit default value `0` is applied to a member (`@default(0)`), this function should return true in that case?